vdr/didservice/resolvers.go

```diff
@@ -164,16 +168,39 @@ func (r KeyResolver) ResolveRelationKey(keyID string, validAt *time.Time, relati
 		ResolveTime: validAt,
 	})
 	if err != nil {
-		return "", err
+		return nil, err
+	}
+	relationships, err := resolveRelationships(doc, relationType)
+	if err != nil {
+		return nil, err
 	}
-	relationships, _ := resolveRelationships(doc, relationType)
-
 	for _, rel := range relationships {
 		if rel.ID.String() == keyID {
 			return rel.PublicKey()
 		}
 	}
-	return "", types.ErrKeyNotFound
+	return nil, types.ErrKeyNotFound
+}
+
+func (r KeyResolver) ResolveKey(id did.DID, validAt *time.Time, relationType types.RelationType) (ssi.URI, crypto.PublicKey, error) {
+	doc, _, err := r.Store.Resolve(id, &types.ResolveMetadata{
+		ResolveTime: validAt,
+	})
+	if err != nil {
+		return ssi.URI{}, nil, err
+	}
+	keys, err := resolveRelationships(doc, relationType)
+	if err != nil {
+		return ssi.URI{}, nil, err
+	}
+	if len(keys) > 0 {
```

Comment:
if len == 0 return err?

Reply:
I don't have a particular preference, but I'll swap it around.